### DISCUSSION TITLE
Which BMS are you using with dbus-serialbattery?

### DISCUSSION THREAD

**Discussion by mr-manuel** [Louisvdw/dbus-serialbattery#546] [General]
To get a little overview which BMS are used by the community **with this driver** we ask you to participate to this poll :-)

Since polls are a little restricted in discussions and the options are limited we are doing it this way.

## Please upvote for the BMS you are using with this driver

This can simply be done by pressing the arrow up under your BMS name.

![grafik](https://github.com/Louisvdw/dbus-serialbattery/assets/12592765/098e3c57-b550-45b2-9258-e61dbc9c3e23)

**DO NOT ADD A NEW COMMENTS, BUT ONLY REPLIES UNDER THE EXISTING BMS NAMES**

## Testing

Since we do not have all BMS for testing, we need as many people per BMS to test pre-release drivers. If you are available to test changes/improvements for your driver, please write a **reply** under the BMS that you have with "available to test".

**Comment by mr-manuel** [comment]
ANT

**Reply by dmorgado18** [reply]
I'm using it on a JBD cable and everything goes well.
I wanted to add an Ant Bms via bluetooth, but I can't.
I have an emus BMS with PC cable to configure it on the computer and additionally it has a G1 Smartphone Connectivity Module. Possible to work with your script

**Comment by mr-manuel** [comment]
Daly

**Reply by transistorgit** [reply]
2 parallel Daly Smart BMS 100A 16s
HD Version BMS-ST103-309E
Active Balancers and parallel Module

additionally I have 2 WNT Boards that I currently don't use, because the CAN support was so buggy.

**Reply by LundSoftwares** [reply]
4s Sinowealth version

**Reply by transistorgit** [reply]
available to test 😎

**Reply by towatai** [reply]
2 parallel Daly Smart BMS 150A, 8s connected via RS485. The UART-Connection seems much slower and i had sporadic, short disconnects (The holy "error #67"). Since i use the RS485-Cables, everthing seems to work fine.
Each battery pack has a Heltec Active Balancer, which can be switched on and off via a Shelly 1 (via NodeRed) depending on the battery voltage. So the thing does not run all the time.

<a href="https://abload.de/image.php?img=pxl_20230415_22191175y4cdm.jpg"><img src="https://abload.de/thumb/pxl_20230415_22191175y4cdm.jpg" /></a>

**Reply by ofernander** [reply]
@mr-manuel using a single 4S 200A Daly BMS on your latest Beta release and after some custom config it's working great! Really appreciate all the input you've given to this project.

**Reply by RoDeP0** [reply]
Daly Sinowealth on a 200 Ah custom build. Worked flawlessly last year on VenusOS 2.8, yet have to get it working on 3.2x (RPI 3B+). Documentation says "Disabled by default as it is causing issues". What are those issues?

**Reply by Louisvdw** [reply]
@RoDeP0 the Sinowealth comunications protocol is very basic and as such if we enable the driver for it most other BMS also respond on those commands and are incorrectly picked up as being a Sinowealth BMS. The rest of the calls thein fail. 
So if you are using a Sinowealth and want to upgrade to the latest version you will need to enable the Sinowealth driver and disable all the others. And then it should work like before.

**Reply by janekx** [reply]
3x 300A on one install and 1x 300A on diferent install

**Comment by mr-manuel** [comment]
ECS

**Comment by mr-manuel** [comment]
JBD/LLT

**Reply by Jaco-TTT** [reply]
JBD, the new model, has an active balancer added, all pre-wired.

In the older versions, adding a Neey Active Balancer seems to be a very good addition according to Ande from the Off-Grid Garage.

**Reply by packrat1969** [reply]
What is the model number for this JBD BMS with active balancing?

**Reply by wherzig** [reply]
My JBD BMS (JBD = JiaBaiDa with active Balancer 160mA SP22S003B) is not working with Version 1.0 and higher. It is only recognized with Version 0.14.3 . Very strange.

**Reply by sledge2019** [reply]
Using the 1.0 Version together with a JBD-AP20S006-L20S-200A-B and a NEEY active Balancer on an 18S battery. Runs like a charme.

**Reply by pmarches** [reply]
Using v1.0.20230531 with two centerpower LifePo4, 12V, 200Ah.

**Reply by RobertGiroux** [reply]
JBD in Bassen Green 230A 24V (200A BMS) drop ins.

**Comment by mr-manuel** [comment]
JKBMS

**Reply by bellancal** [reply]
Need the active balancing!

**Reply by ogurevich** [reply]
2 pieces :)

**Reply by Ahjwinth** [reply]
At the moment 1x 8s mobilehome battery, in the near future 4x16S for ESS at Home.

**Reply by peregrines2** [reply]
Using 4 JK BMS with active balancing

**Reply by AncoDiara** [reply]
Have several JKbms's active some 4S 8S and 14S. mostly interested in testing now via bluetooth serial connection to venusos, due to time/priority constraints, I plan testing bluetooth with the serial driver in the coming weeks.

**Reply by AncoDiara** [reply]
Way cool! Well done you programmers. JKbms-BLE working great on Ekrano gx! used the nightly build for helping testing. now discharging the battery and next plan write a Node red script to do automatic battery capacity testing. For example test a battery 3 runs of charge and discharge.

**Reply by Weifel53** [reply]
one 8s with active balancing

**Reply by Beve-dev** [reply]
JKBMS B2A8S20P connected with RS485
1x  Hardware: V9.X Software: V9.08W
1x  Hardware: V11.XW Software: V11.26
works now

**Reply by 1234enough** [reply]
Working great with jkbms for 1yr, just updated and gets even better, great coding 👍

**Reply by newegnaro** [reply]
JK Bms with 16S 230ah ESS
Happy to test with a bit of support regarding backing out test drivers.

**Reply by AncoDiara** [reply]
B2A20S20P is working great via bluetooth. However I do disable "False" most of the extra options as on bigger battery packs NMC the extra control options don't work yet as I was hoping for. Need to spend more time in testing/tweaking those options to learn them.
I also found the Jk balancer and 8s bms's 100amps and lower are not recognized by the bluetooth driver despite filling in the correct mac address which start with 20: instead of the C8:. How can I help make them work? Can I order you one of those or pay for them and you order your self an 8s 40amp jk bms or  the jk balancer or both?

**Reply by rolandlueders** [reply]
Three Banks 320Ah in parallel with JKB2A20S20P via RS485 with Battery aggregator ist working well. Serialbattery 1.1.20240128beta

**Reply by Mike-Bou** [reply]
2 Jkbms  2A bal, 4S-8S bms. Lifepo4 24v 320Ah & 12v 310Ah. Raspberrypi3's. One works great. Other cannot receive anything on the ttyUSB1 . Cable ok when swapped. I do not know how to find the Bluetooth MAC Address on a JK BMS???

**Reply by MatKra89** [reply]
3 Jkbms B2A20S20P @ 16S 280Ah LFP, Raspberrypi 3B and connected via the pi's internal BLUETOOTH!
Works well, but some problems with SoC reset. A newer version (nightly build) is able to reset all of the JK´s, but the last stable (v1.2.20240408) is not. The nightly build has problems with the correct CVL for me, it always goes up to the limit of the bms, i.e. way too high. The sporadic transmission errors with individual cell voltages being too high/low are also gone in the nightly build, that's fine and I'm looking forward to a new version. 
Big praise from the YouTube channel Energie&Hobby at this point.

**Reply by jjdejong** [reply]
BD6A20S10P

**Reply by noonscoomo** [reply]
2 x JK BMS @ 16S 230Ah LFP on Cerbo GX via RS485. 
Had trouble to learn first to set the cell voltages in settings.py _before_ trying to set a guess for "charge" since at LFP the charge value always resets to 100% because of the default low max voltage setting.

**Reply by Pim57** [reply]
BD6A20S15P Bluetooth to Cerbo GX,SOC reset works great

**Reply by Baxter117** [reply]
- 3x JK over BLE in System1 ( 3x 16S )

- 2x JK over BLE in System2 ( 2x 18S )

- 1x JK over BLE in System3 (1x 16S )

hope BLE connection will be improved in future :)

**Comment by mr-manuel** [comment]
Renogy

**Reply by mark-1003** [reply]
Using two Renogy Batteries/BMS's.
Each one connected with an Rs485 interface.

**Reply by BlueY21** [reply]
Hello mark-1003,
I also have Renogy Smart Batteries connected in parallel and am in the process of connecting them to a Cerbo GX.  I understand that presently, they each need to be connected separately :-(  
I have Okiwan RJ45-USB Serial Adapters but I think the signal pin positions on their RJ45 male connectors do not match the signal pin positions on the Renogy battery RJ45 sockets.  (The Okiwan Adapters and all the similar products are intended to connect PCs to data communications equipment console ports,)  I'm planning on making a cat6-type adapter cable to transpose the Tx,  Rx, and Gnd signals and omit the extraneous rs232 control signals provided by the Okiwan adapter.  
How did you connect your batteries to your Venus device?  
What type of RJ45-USB Serial Adapters did you use?  
What signals and connector pins on the batteries did you use?   
Is it required to use the UpLink socket on the Renogy batteries, vs. the DnLink socket?  
I currently use the Renogy BT-2 Bluetooth Dongle.  Can that still be used if the Batteries are connected to the Cerbo? 
I plan on putting the dbus-serial battery .tar file and a config.ini file on a micro-sd chip and put then leave it pluged into the Cerbo.  Is that what you did?
For the 100Ah batteries, I plan to set the MAX_BATTERY_CHARGE_CURRENT to 20 (=0.2C) and the MAX_BATTERY_DISCHARGE_CURRENT to 100 in the config.ini.  Would this be correct?  
Have you considered any of the Battery Aggregator additional drivers?

Your advice would be greatly appreciated.

**Reply by gambleben** [reply]
I'm using the Renogy USB RS485 to USB adapter and the Renogy BT-2 Bluetooth Dongle concurrently. I did the original development on the Renogy driver but haven't contributed anything in quite awhile. At one time I had a working fork that aggregated all four of my Renogy batteries to make them look like one massive battery to dbus-serialbattery, but I haven't kept the code up to date and need to re-implement that for the latest version of VenusOS and this driver, ideally in a more elegant fashion along with refactoring to use modbus and pull in all of the alarms now that we have full documentation of the protocol.
One thing to note is that when you plug in the Renogy bluetooth, it assigns new MODBUS addresses to all of the connected batteries. New from the factory I think they default to 254 or something like that, but after connecting the BT-2 module they will start at address 50 and count up from there.

**Reply by jaa2261** [reply]
I have a bank of 8 Renogy RBT100LFP12S batteries. Happy to help with testing. 
@gambleben how do you get the RS485 and BT-2 to operate concurrently? Do you use the Renogy Hub or a RS485 splitter?

**Comment by mr-manuel** [comment]
TIAN/LIFE Power

**Reply by Justice12t** [reply]
I think most of the work for the 12v battery has been gathered.

**Reply by Justice12t** [reply]
Available to test

**Reply by LilDucky** [reply]
I have a Tian Power 48V BMS that is 15 cell and badged as OptimumNano. This driver almost works, and I had to reduce the number of temp sensors to get rid of an out of range error. The driver could really use a refactor to correctly implement the group number byte and the following length byte.

Another improvement would be to implement the daisy chained module addressing to save on ports for connection.

Edited to add: found a python lib that may make this work a ton easier. 

https://github.com/Frankkkkk/python-pylontech

I will try but my python skills are horrendous.

**Comment by mr-manuel** [comment]
Seplos

**Reply by tnt760** [reply]
Seplos v3 here

**Comment by mr-manuel** [comment]
Overkill Solar

**Reply by GitHubLex14** [reply]
I use: 3x Overkill Solar BMS (16s 100A) with 3x USB Module for BMS. Considered as SerialBattery(LLT/JBD) at victron
3 sets of 16 cells LifePo4 280Ah. Is 3x280Ah=840Ah total.
